foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/indirection/UnitOfWorkValueHolder.java

```diff
@@ -162,17 +163,18 @@ protected Object instantiateImpl() {
                         return wrapped.instantiateForUnitOfWorkValueHolder(this);
                     }
                 }
-            }
-            if (!this.wrappedValueHolder.isInstantiated()){
-                //if not instantiated then try and load the UOW versions to prevent the whole loading from the cache and cloning
-                //process
-                Object result = ((DatabaseValueHolder)this.wrappedValueHolder).getValue((UnitOfWorkImpl) this.session);
-                if (result != null){
-                    return result;
+                if (!((DatabaseValueHolder)this.wrappedValueHolder).isInstantiated()){
```

Comment:
is this explicit cast required here? If so, would it make sense to store/cache the result of the cast in some local field and reuse it a line or two later?